examples/weather.rs

```diff
@@ -0,0 +1,134 @@
+	ws_server::WsServer,
+	ws_server::RpcContextModule,
+};
+
+#[derive(Deserialize, Serialize, Debug, Default, PartialEq)]
+struct Weather {
+	name: String,
+	wind: Wind,
+	clouds: Clouds,
+	main: Main,
+}
+#[derive(Deserialize, Serialize, Debug, Default, PartialEq)]
+struct Clouds {
+	all: usize,
+}
+#[derive(Deserialize, Serialize, Debug, Default, PartialEq)]
+struct Main {
+	temp: f64,
+	pressure: usize,
+	humidity: usize,
+}
+#[derive(Deserialize, Serialize, Debug, Default, PartialEq)]
+struct Wind {
+	speed: f64,
+	deg: usize,
+}
+
+
+impl RestPath<&(String, String)> for Weather {
+	fn get_path(params: &(String, String)) -> Result<String, RestsonError> {
+		const API_KEY: &'static str = "f6ba475df300d5f91135550da0f4a867";
+		Ok(
+			String::from(
+				format!("data/2.5/weather?q={}&units={}&appid={}",
+					params.0,
+					params.1,
+					API_KEY,
+				)
+			)
+		)
+	}
+}
+
+/// Example of setting up a subscription that polls a remote API, in this case the api.openweathermap.org/weather, and
+/// sends the data back to the subscriber whenever the weather in London changes.
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+	env_logger::init();
+	let addr = run_server().await?;
+	let url = format!("ws://{}", addr);
+
+	let client = WsClientBuilder::default().build(&url).await?;
+
+	// Subscription to the london weather
+	let params = JsonRpcParams::Array(vec!["London,uk".into(), "metric".into()]);
+	let mut weather_sub = client.subscribe::<Weather>("weather_sub", params, "weather_unsub").await?;
+	while let Some(w) = weather_sub.next().await {
+		println!("[client] London weather: {:?}", w);
+	}
+
+	Ok(())
+}
+
+struct WeatherApiCx {
+	api_client: restson::RestClient,
+	last_weather: Weather,
+}
+
+async fn run_server() -> anyhow::Result<SocketAddr> {
+	let mut server = WsServer::new("127.0.0.1:0").await?;
+
+	let api_client = restson::RestClient::new("http://api.openweathermap.org").unwrap();
+	let last_weather = Weather::default();
+	let cx = Mutex::new(WeatherApiCx { api_client, last_weather });
+	let mut module = RpcContextModule::new(cx);
+	module
+		.register_subscription_with_context("weather_sub", "weather_unsub", |params, sink, cx| {
+			let params: (String, String) = params.parse()?;
+			log::debug!(target: "server", "subscribed with params={:?}", params);
+			std::thread::spawn(move || loop {
+				let mut cx = cx.lock().unwrap();
+				let current_weather: Weather = cx.api_client.get(&params).unwrap();
+				if current_weather != cx.last_weather {
+					log::debug!(target: "server", "Fetched London weather: {:?}, sending", current_weather);
+					sink.send(&current_weather).expect("Sending should work yes?");
+					cx.last_weather = current_weather;
+				} else {
+					log::trace!(target: "server", "Same weather as before")
+				}
+				std::thread::sleep(std::time::Duration::from_millis(500));
+			});
+			Ok(())
+		})
+		.unwrap();
+
+	server.register_module(module.into_module()).unwrap();
```

Comment:
@niklasad1 I think this is a bit awkward, having to cast my context module into a module to register it with the server. Would dynamic dispatch be possible here you think, so that `register_module` takes a `dyn RpcModuleTraitObject` instead?

Reply:
I think we should merge `RpcModule` and `RpcContextModule` with `NoopContext` and just have a type def:

```
pub struct RpcContextModule<Context> {
...
}

pub type RpcModule: RpcContextModule<()>;

```

I think that would fix it without having do trait objects

Reply:
That's a good idea, I'll explore that.